obspy/io/nlloc/core.py

```diff
@@ -138,21 +137,20 @@ def read_nlloc_hyp(filename, coordinate_converter=None, picks=None, **kwargs):
     signature, version, date, time = line.rsplit(" ", 3)
     creation_time = UTCDateTime().strptime(date + time, str("%d%b%Y%Hh%Mm%S"))
 
-    # maximum likelihood origin location info line
-    line = lines["HYPOCENTER"]
-
-    x, y, z = map(float, line.split()[1:7:2])
-
-    if coordinate_converter:
-        x, y, z = coordinate_converter(x, y, z)
-
-    # origin time info line
+    # maximum likelihood origin time and location lon lat info line
     line = lines["GEOGRAPHIC"]
 
     year, month, day, hour, minute = map(int, line.split()[1:6])
     seconds = float(line.split()[6])
     time = UTCDateTime(year, month, day, hour, minute, seconds)
 
+    x, y, z = map(float, line.split()[8:13:2])
+
+    if coordinate_converter:
+        # maximum likelihood origin location in km info line
+        line = lines["HYPOCENTER"]
+        x, y, z = coordinate_converter(*map(float, line.split()[1:7:2]))
+
```

Comment:
Sorry if it seems pedantic, but I still think it is logically easier to read like..
```python
    line = lines["GEOGRAPHIC"]
    year, month, day, hour, minute = map(int, line.split()[1:6])
    seconds = float(line.split()[6])
    time = UTCDateTime(year, month, day, hour, minute, seconds)

    if coordinate_converter:
        # maximum likelihood origin location in km info line
        line = lines["HYPOCENTER"]
        x, y, z = coordinate_converter(*map(float, line.split()[1:7:2])) 
    else:
        line = lines["GEOGRAPHIC"]
        x, y, z = map(float, line.split()[8:13:2])
```